pretyping/vnorm.ml

```diff
@@ -282,10 +282,12 @@ and nf_stk ?from:(from=0) env sigma c t stk  =
       let (mib,mip) = Inductive.lookup_mind_specif env ind in
       let nparams = mib.mind_nparams in
       let params,realargs = Util.Array.chop nparams allargs in
-      let nparamdecls = Context.Rel.length (Inductive.inductive_paramdecls (mib,u)) in
-      let pT =
-        hnf_prod_applist_decls env nparamdecls (type_of_ind env (ind,u)) (Array.to_list params) in
-      let pctx, p, relevance = nf_predicate env sigma (ind,u) mip params [] (type_of_switch sw) pT in
+      let pctx =
+        let realdecls, _ = List.chop mip.mind_nrealdecls mip.mind_arity_ctxt in
+        let nas = Array.of_list @@ (List.rev_map RelDecl.get_annot realdecls @ [annotR (Name (Id.of_string "c"))]) in
+        expand_arity (mib, mip) (ind, u) params nas
```

Comment:
Just a few nitpicks because this code hurts my eyes.

First, there is not much point in using the `@@` operator if you are parenthesizing the argument anyway. Also, it would have been more readable to move `Array.of_list` to the `expand_arity` line. Finally, `annotR (Name ...)` can be advantageously replaced by `nameR`.

```ocaml
        let nas = List.rev_map RelDecl.get_annot realdecls @ [nameR (Id.of_string "c")] in
        expand_arity (mib, mip) (ind, u) params (Array.of_list nas)
```

Reply:
Done.